iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusTerminal.java

```diff
@@ -82,15 +82,19 @@ public MergedBus getConnectableBus() {
     }
 
     void setConnectableBusId(String connectableBusId) {
-        this.connectableBusId.set(network.get().getVariantIndex(), connectableBusId);
+        int variantIndex = network.get().getVariantIndex();
+        this.connectableBusId.set(variantIndex, connectableBusId);
```

Comment:
I don't understand why you change this? Do you forget to add the notification here?